core/state_processor.go

```diff
@@ -119,8 +119,15 @@ func ApplyTransaction(config *ChainConfig, bc *BlockChain, gp *GasPool, statedb
 	}
 
 	// Update the state with pending changes
+	var root []byte
+	if config.IsAtlantis(header.Number) {
+		statedb.Finalise(true)
+	} else {
+		root = statedb.IntermediateRoot(config.IsAtlantis(header.Number)).Bytes()
```

Comment:
IFF `isAtlantis === true` this code block won't get hit, you can probably just default it to false:
```suggestion
		root = statedb.IntermediateRoot(false).Bytes()
```

Reply:
Future proofing since putting false is logically false, as state trie clearing should only occur when `IsAtlantis` is true. Is the same logically to put false in this case, but if intermediate root is needed in the future, I think it's better to be explicit.